tests/ZiraLink.Api.Application.UnitTests/Services/CustomerServiceTest.cs

```diff
@@ -89,52 +91,310 @@ public async Task CreateLocallyAsync_WhenEverythingIsOk_ShouldBeSucceeded(string
+        {
+            var mockHttpTools = new Mock<IHttpTools>();
+
+            Assert.True(string.IsNullOrWhiteSpace(password));
+            CustomerService customerService = new CustomerService(_testTools.AppMemoryDbContext, mockHttpTools.Object);
+
+            var exception = await Assert.ThrowsAsync<ArgumentNullException>(() => customerService.CreateAsync(username, password, email, name, family, CancellationToken.None));
+            Assert.Equal("password", exception.ParamName);
+        }
+
+        [Theory]
+        [InlineData("TestUser", "TestPassword", "", "Test", "Testi")]
+        public async Task CreateCustomer_WhenEmailIsEmpty_ShouldBeFailed(string username, string password, string email, string name, string family)
+        {
+            var mockHttpTools = new Mock<IHttpTools>();
+
+
+            Assert.True(string.IsNullOrWhiteSpace(email));
+            CustomerService customerService = new CustomerService(_testTools.AppMemoryDbContext, mockHttpTools.Object);
+
+            var exception = await Assert.ThrowsAsync<ArgumentNullException>(() => customerService.CreateAsync(username, password, email, name, family, CancellationToken.None));
+            Assert.Equal("email", exception.ParamName);
+        }
+
+        [Theory]
+        [InlineData("TestUser", "TestPassword", "TestEmail@email.com", "", "Testi")]
+        public async Task CreateCustomer_WhenNameIsEmpty_ShouldBeFailed(string username, string password, string email, string name, string family)
+        {
+            var mockHttpTools = new Mock<IHttpTools>();
+
+            Assert.True(string.IsNullOrWhiteSpace(name));
+            CustomerService customerService = new CustomerService(_testTools.AppMemoryDbContext, mockHttpTools.Object);
+
+            var exception = await Assert.ThrowsAsync<ArgumentNullException>(() => customerService.CreateAsync(username, password, email, name, family, CancellationToken.None));
+            Assert.Equal("name", exception.ParamName);
+        }
+
+        [Theory]
+        [InlineData("TestUser", "TestPassword", "TestEmail@email.com", "Test", "")]
+        public async Task CreateCustomer_WhenFamilyIsEmpty_ShouldBeFailed(string username, string password, string email, string name, string family)
+        {
+            var mockHttpTools = new Mock<IHttpTools>();
+
+            Assert.True(string.IsNullOrWhiteSpace(family));
+            CustomerService customerService = new CustomerService(_testTools.AppMemoryDbContext, mockHttpTools.Object);
+
+            var exception = await Assert.ThrowsAsync<ArgumentNullException>(() => customerService.CreateAsync(username, password, email, name, family, CancellationToken.None));
+            Assert.Equal("family", exception.ParamName);
+        }
+
+        [Theory]
+        [InlineData("TestUser2", "TestPassword", "TestUser2@ZiraLink.com", "Test", "Testi")]
```

Comment:
I suggest replacing the email with TestEmail@ziralink.local

Reply:
Done.